<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started with immunedeconv • immunedeconv</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with immunedeconv">
<meta property="og:description" content="immunedeconv">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">immunedeconv</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/immunedeconv.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/detailed_example.html">Detailed Usage Example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/icbi-lab/immunedeconv/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="immunedeconv_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="immunedeconv_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="immunedeconv_files/viz-1.8.2/viz.js"></script><link href="immunedeconv_files/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="immunedeconv_files/grViz-binding-1.0.6.1/grViz.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started with immunedeconv</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/icbi-lab/immunedeconv/blob/master/vignettes/immunedeconv.Rmd"><code>vignettes/immunedeconv.Rmd</code></a></small>
      <div class="hidden name"><code>immunedeconv.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Methods like fluorescence activated cell sorting (FACS) or Immunohistochemistry (IHC)-staining have been used as a gold standard to estimate the immune cell content within a sample, however these methods are limited in their scalability and by the availability of good antibodies against the cell type markers. High throughput transcriptomic methods allow to get a transcriptional landscape in the sample with a relatively small amount of material that can be extremely limited in clinical settings (e.g. tumor biopsies), which led to high utility of methods like RNA-seq and microarrays to characterize patient tumor samples. However, RNA-seq does not provide a detailed information on a cellular composition of a sample, which then has to be inferred using computational techniques.</p>
<div id="conceptual-differences-between-the-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#conceptual-differences-between-the-methods" class="anchor"></a>Conceptual differences between the methods</h2>
<p>Such methods can, in general, be classified in two categories:</p>
<ul>
<li>Marker gene-based approaches and</li>
<li>deconvolution-based approaches.</li>
</ul>
<div class="figure">
<img src="img/concepts_deconvolution.gif" alt=""><p class="caption"><em>Fig by <span class="citation">Finotello and Trajanoski (<a href="#ref-Finotello2018" role="doc-biblioref">2018</a>)</span>.</em></p>
</div>
<p>Marker gene based approaches (a) are based on a list of genes (signature), that are characteristic for a cell type. By looking at the expression values of signature genes, every cell type is quantified independently, either using the gene expression values directly (MCP-counter) or by performing a statistical test for enrichment of the signatures (xCell).</p>
<p>Deconvolution methods (b) formulate the problem as a system of equations that describe the gene expression of a sample as the weighted sum of the contributions of the different cell types. By solving the inverse problem, cell type fractions can be inferred given a signature matrix and the mixed gene expression. This can be accomplished using <span class="math inline">\(\nu\)</span>-Support Vector Regression (SVR) (CIBERSORT) constrained least square regression (quanTIseq, EPIC) or linear least square regression (TIMER).</p>
<p>For more information, check out the review by <span class="citation">Finotello and Trajanoski (<a href="#ref-Finotello2018" role="doc-biblioref">2018</a>)</span>.</p>
</div>
</div>
<div id="run-the-deconvolution" class="section level1">
<h1 class="hasAnchor">
<a href="#run-the-deconvolution" class="anchor"></a>Run the deconvolution</h1>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>Input Data</h2>
<p>The input data is a <code>gene</code> <span class="math inline">\(\times\)</span> <code>sample</code> gene expression matrix. In general values should be</p>
<ul>
<li>TPM-normalized</li>
<li>
<em>not</em> log-transformed.</li>
</ul>
<p>For xCell and MCP-counter this is not so important. xCell works on the ranks of the gene expression only and MCP-counter sums up the gene expression values.</p>
<p>Rownames are expected to be HGNC gene symbols. Instead of a matrix, <em>immunedeconv</em> also supports ExpressionSets (<a href="#using-expressionsets">see below</a>).</p>
</div>
<div id="run-a-method" class="section level2">
<h2 class="hasAnchor">
<a href="#run-a-method" class="anchor"></a>Run a method</h2>
<p>This package gives you easy access to these methods. To run a method with default options, simply invoke</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">immunedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="va">gene_expression_matrix</span>, <span class="va">method</span><span class="op">)</span></code></pre></div>
<p>where <code>gene_expression_matrix</code> is a matrix with genes in rows and samples in columns. The rownames must be HGNC symbols and the colnames must be sample names. The method can be one of</p>
<pre><code>quantiseq
timer
cibersort
cibersort_abs
mcp_counter
xcell
epic</code></pre>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
<p>For this example, we use a dataset of four melanoma patients from <span class="citation">Racle et al. (<a href="#ref-EPIC2017" role="doc-biblioref">2017</a>)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="fu">immunedeconv</span><span class="fu">::</span><span class="va"><a href="../reference/dataset_racle.html">dataset_racle</a></span><span class="op">$</span><span class="va">expr_mat</span>, <span class="st">"quantiseq"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">res</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">cell_type</th>
<th align="right">LAU125</th>
<th align="right">LAU355</th>
<th align="right">LAU1255</th>
<th align="right">LAU1314</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">B cell</td>
<td align="right">0.02</td>
<td align="right">0.43</td>
<td align="right">0.02</td>
<td align="right">0.49</td>
</tr>
<tr class="even">
<td align="left">Macrophage M1</td>
<td align="right">0.01</td>
<td align="right">0.00</td>
<td align="right">0.03</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Macrophage M2</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.01</td>
</tr>
<tr class="even">
<td align="left">Monocyte</td>
<td align="right">0.18</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Neutrophil</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.19</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td align="left">NK cell</td>
<td align="right">0.05</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">T cell CD4+ (non-regulatory)</td>
<td align="right">0.01</td>
<td align="right">0.44</td>
<td align="right">0.00</td>
<td align="right">0.38</td>
</tr>
<tr class="even">
<td align="left">T cell CD8+</td>
<td align="right">0.00</td>
<td align="right">0.03</td>
<td align="right">0.09</td>
<td align="right">0.05</td>
</tr>
<tr class="odd">
<td align="left">T cell regulatory (Tregs)</td>
<td align="right">0.02</td>
<td align="right">0.10</td>
<td align="right">0.06</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">Myeloid dendritic cell</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">uncharacterized cell</td>
<td align="right">0.71</td>
<td align="right">0.00</td>
<td align="right">0.61</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
</div>
<div id="special-case-cibersort" class="section level3">
<h3 class="hasAnchor">
<a href="#special-case-cibersort" class="anchor"></a>special case: CIBERSORT</h3>
<p>CIBERSORT is only freely available for academic users and could not be directly included in this package. To use CIBERSORT with this package, you need to register on the <a href="https://cibersort.stanford.edu/">cibersort website</a>, obtain a license, and download the CIBERSORT source code.</p>
<p>The source code package contains two files, that are required:</p>
<pre><code>CIBERSORT.R
LM22.txt</code></pre>
<p>Note the storage location of these files. When using <code>immunedeconv</code>, you need to tell the package where it can find those files:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://icbi-lab.github.io/immunedeconv">immunedeconv</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/set_cibersort_binary.html">set_cibersort_binary</a></span><span class="op">(</span><span class="st">"/path/to/CIBERSORT.R"</span><span class="op">)</span>
<span class="fu"><a href="../reference/set_cibersort_mat.html">set_cibersort_mat</a></span><span class="op">(</span><span class="st">"/path/to/LM22.txt"</span><span class="op">)</span></code></pre></div>
<p>Afterwards, you can call</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="va">your_mixture_matrix</span>, <span class="st">"cibersort"</span><span class="op">)</span>   <span class="co"># or 'cibersort_abs'</span></code></pre></div>
<p>as for any other method.</p>
</div>
<div id="special-case-timer" class="section level3">
<h3 class="hasAnchor">
<a href="#special-case-timer" class="anchor"></a>special case: TIMER</h3>
<p>TIMER uses indication-specific reference profiles. Therefore, you <em>must</em> specify the tumor type when running TIMER:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="va">your_mixture_matrix</span>, <span class="st">"timer"</span>,
            indications<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SKCM"</span>, <span class="st">"SKCM"</span>, <span class="st">"BLCA"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><code>indications</code> needs to be a vector that specifies an indication for each sample (=column) in the mixture matrix. The indications supported by TIMER are</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">immunedeconv</span><span class="fu">::</span><span class="va"><a href="../reference/timer_available_cancers.html">timer_available_cancers</a></span></code></pre></div>
<pre><code>##  [1] "kich" "blca" "brca" "cesc" "gbm"  "hnsc" "kirp" "lgg"  "lihc" "luad"
## [11] "lusc" "prad" "sarc" "pcpg" "paad" "tgct" "ucec" "ov"   "skcm" "dlbc"
## [21] "kirc" "acc"  "meso" "thca" "uvm"  "ucs"  "thym" "esca" "stad" "read"
## [31] "coad" "chol"</code></pre>
<p>What the abbreviations stand for is documented on the <a href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/tcga-study-abbreviations">TCGA wiki</a>.</p>
</div>
<div id="using-expressionsets" class="section level3">
<h3 class="hasAnchor">
<a href="#using-expressionsets" class="anchor"></a>Using ExpressionSets</h3>
<p>The <a href="https://bioconductor.org/">Bioconductor</a> <a href="https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/ExpressionSet">ExpressionSet</a> is a convenient way to store a gene expression matrix with metadata for both samples and genes in a single object.</p>
<p><code>immunedeconv</code> supports the use of an ExpressionSet instead of a gene expression matrix. In that case, <code>pData</code> requires a column that contains gene symbols. Which one needs to be specified in the <code><a href="../reference/deconvolute.html">deconvolute()</a></code> call:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="va">my_expression_set</span>, <span class="st">"quantiseq"</span>, column <span class="op">=</span> <span class="st">"&lt;column name&gt;"</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="cell-type-re-mapping" class="section level1">
<h1 class="hasAnchor">
<a href="#cell-type-re-mapping" class="anchor"></a>Cell-type re-mapping</h1>
<p>To provide consistently named results independent of the method, we defined a <em>controlled vocabulary</em> (CV) of cell-types and arranged them in a tree.</p>
<div id="htmlwidget-f0cb8ca2ef93db8e8b8c" style="width:700px;height:432.632880098888px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-f0cb8ca2ef93db8e8b8c">{"x":{"diagram":"digraph {\n\ngraph [rankdir = \"LR\"]\n\n\n\n  \"1\" [label = \"root2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: root2\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"2\" [label = \"cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"3\" [label = \"immune cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"4\" [label = \"Hematopoietic stem cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"5\" [label = \"lymphoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"6\" [label = \"B cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"7\" [label = \"B cell memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"8\" [label = \"B cell naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"9\" [label = \"B cell plasma\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"10\" [label = \"Class-switched memory B cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"11\" [label = \"Common lymphoid progenitor\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"12\" [label = \"NK cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"13\" [label = \"NK cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"14\" [label = \"NK cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"15\" [label = \"Plasmacytoid dendritic cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"16\" [label = \"T cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"17\" [label = \"T cell CD4+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"18\" [label = \"T cell CD4+ (non-regulatory)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"19\" [label = \"T cell CD4+ Th1\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"20\" [label = \"T cell CD4+ Th2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"21\" [label = \"T cell CD4+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"22\" [label = \"T cell CD4+ memory activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"23\" [label = \"T cell CD4+ memory resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"24\" [label = \"T cell CD4+ central memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"25\" [label = \"T cell CD4+ effector memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"26\" [label = \"T cell CD4+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"27\" [label = \"T cell follicular helper\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"28\" [label = \"T cell regulatory (Tregs)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"29\" [label = \"T cell CD8+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"30\" [label = \"T cell CD8+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"31\" [label = \"T cell CD8+ central memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"32\" [label = \"T cell CD8+ effector memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"33\" [label = \"T cell CD8+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"34\" [label = \"T cell gamma delta\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"35\" [label = \"T cell NK\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"36\" [label = \"myeloid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"37\" [label = \"Basophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"38\" [label = \"Common myeloid progenitor\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"39\" [label = \"Granulocyte-monocyte progenitor\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"40\" [label = \"Eosinophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"41\" [label = \"Mast cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"42\" [label = \"Mast cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"43\" [label = \"Mast cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"44\" [label = \"Monocytic lineage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"45\" [label = \"Macrophage/Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"46\" [label = \"Macrophage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"47\" [label = \"Macrophage M0\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"48\" [label = \"Macrophage M1\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"49\" [label = \"Macrophage M2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"50\" [label = \"Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"51\" [label = \"Myeloid dendritic cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"52\" [label = \"Myeloid dendritic cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"53\" [label = \"Myeloid dendritic cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"54\" [label = \"Neutrophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"55\" [label = \"other cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"56\" [label = \"Cancer associated fibroblast\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"57\" [label = \"cancer cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"58\" [label = \"Melanoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"59\" [label = \"Ovarian carcinoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"60\" [label = \"Endothelial cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"61\" [label = \"uncharacterized cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"62\" [label = \"score\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"63\" [label = \"cytotoxicity score\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"64\" [label = \"immune score\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"65\" [label = \"microenvironment score\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"66\" [label = \"stroma score\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n\"1\"->\"2\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"1\"->\"62\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"3\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"55\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"62\"->\"63\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"62\"->\"64\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"62\"->\"65\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"62\"->\"66\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"4\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"5\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"36\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"55\"->\"56\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"55\"->\"57\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"55\"->\"60\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"55\"->\"61\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"6\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"11\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"12\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"15\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"16\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"37\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"38\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"39\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"40\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"41\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"44\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"36\"->\"54\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"57\"->\"58\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"57\"->\"59\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"7\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"8\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"9\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"10\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"12\"->\"13\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"12\"->\"14\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"16\"->\"17\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"16\"->\"29\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"16\"->\"34\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"16\"->\"35\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"41\"->\"42\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"41\"->\"43\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"44\"->\"45\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"44\"->\"51\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"18\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"28\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"29\"->\"30\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"29\"->\"33\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"45\"->\"46\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"45\"->\"50\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"51\"->\"52\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"51\"->\"53\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"18\"->\"19\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"18\"->\"20\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"18\"->\"21\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"18\"->\"26\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"18\"->\"27\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"31\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"32\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"46\"->\"47\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"46\"->\"48\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"46\"->\"49\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"22\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"23\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"24\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"25\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script><p>For each method, each cell-type is mapped to a node in the tree. If you are curious, it’s all defined in <a href="https://github.com/grst/immunedeconv/blob/master/inst/extdata/cell_type_mapping.xlsx">this excel sheet</a>.</p>
<p>This tree can be used to summarize scores along the tree. For instance, quanTIseq provides scores for regulatory and non-regulatory CD4+ T cells independently, but you are interested in the fraction of overall CD4+ T cells. In that case you can use <code>map_result_to_celltypes</code> to sum up the scores:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="fu">immunedeconv</span><span class="fu">::</span><span class="va"><a href="../reference/dataset_racle.html">dataset_racle</a></span><span class="op">$</span><span class="va">expr_mat</span>, <span class="st">"quantiseq"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/map_result_to_celltypes.html">map_result_to_celltypes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"T cell CD4+"</span><span class="op">)</span>, <span class="st">"quantiseq"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: `funs()` is deprecated as of dplyr 0.8.0.
## Please use a list of either functions or lambdas: 
## 
##   # Simple named list: 
##   list(mean = mean, median = median)
## 
##   # Auto named with `tibble::lst()`: 
##   tibble::lst(mean, median)
## 
##   # Using lambdas
##   list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">res</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">LAU125</th>
<th align="right">LAU355</th>
<th align="right">LAU1255</th>
<th align="right">LAU1314</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">T cell CD4+</td>
<td align="right">0.03</td>
<td align="right">0.54</td>
<td align="right">0.06</td>
<td align="right">0.44</td>
</tr></tbody>
</table>
<p>The algorithm is explained in detail in the methods section of <span class="citation">Sturm et al. (<a href="#ref-sturm2019" role="doc-biblioref">2019</a>)</span>.</p>
</div>
<div id="interpretation-of-scores" class="section level1">
<h1 class="hasAnchor">
<a href="#interpretation-of-scores" class="anchor"></a>Interpretation of scores</h1>
<p>In general, cell-type scores allow for the comparison (1) between samples, (2) between cell-types or (3) both. Between-sample comparisons allow to make statements such as <em>“In patient A, there are more CD8+ T cells than in patient B”</em>. Between-cell-type comparisons allow to make statements such as <em>“In a certain patient, there are more B cells than T cells”</em>. For more information, see our Benchmark paper (<span class="citation">Sturm et al. (<a href="#ref-sturm2019" role="doc-biblioref">2019</a>)</span>).</p>
<div id="methods-that-allow-between-sample-comparisons" class="section level3">
<h3 class="hasAnchor">
<a href="#methods-that-allow-between-sample-comparisons" class="anchor"></a>Methods that allow between-sample comparisons</h3>
<ul>
<li>MCP-counter</li>
<li>xCell</li>
<li>TIMER</li>
</ul>
</div>
<div id="methods-that-allow-between-cell-type-comparisons" class="section level3">
<h3 class="hasAnchor">
<a href="#methods-that-allow-between-cell-type-comparisons" class="anchor"></a>Methods that allow between-cell-type comparisons</h3>
<ul>
<li>CIBERSORT</li>
</ul>
</div>
<div id="methods-that-allow-both" class="section level3">
<h3 class="hasAnchor">
<a href="#methods-that-allow-both" class="anchor"></a>Methods that allow both</h3>
<ul>
<li>EPIC</li>
<li>quanTIseq</li>
<li>CIBERSORT abs. mode</li>
</ul>
<p>EPIC and quanTIseq are currently the only methods providing an absolute score, i.e. a score that can be interpreted as a cell fraction. These methods also provide an estimate for the amount of uncharacterized cells, i.e. cells for that no signature exists. This measure often corresponds to the fraction of cancer cells in the sample.</p>
<p>CIBERSORT abs., while allowing both between- and within-sample comparisons, generates a score in arbitrary units.</p>
</div>
</div>
<div id="faqs" class="section level1">
<h1 class="hasAnchor">
<a href="#faqs" class="anchor"></a>FAQs</h1>
<div id="can-i-specify-a-custom-signature-matrix-through-immunedeconv" class="section level3">
<h3 class="hasAnchor">
<a href="#can-i-specify-a-custom-signature-matrix-through-immunedeconv" class="anchor"></a>Can I specify a custom signature matrix through immunedeconv?</h3>
<p>No, currently not. The reason is that the methods are conceptually different. Some are marker gene based and others deconvolution-based. CIBERSORT performs feature-selection on the matrix while EPIC and quanTIseq don’t. EPIC uses <em>all</em> genes to estimate the inter-sample variance while quanTIseq uses marker genes only. This is also being discussed in <a href="https://github.com/grst/immunedeconv/issues/15">#15</a>.</p>
<p>You can, however, provide custom signatures for most individual methods (see next question).</p>
</div>
<div id="i-want-to-use-a-special-feature-of-a-method-but-i-cannot-access-it-through-the-deconvolute-function-" class="section level3">
<h3 class="hasAnchor">
<a href="#i-want-to-use-a-special-feature-of-a-method-but-i-cannot-access-it-through-the-deconvolute-function-" class="anchor"></a>I want to use a special feature of a method, but I cannot access it through the <code>deconvolute</code> function.</h3>
<p>You can access each method individually through the <code>deconvolute_xxx</code> function. Through these functions you can access all native features. See the <a href="../reference/">function reference</a> for details.</p>
<p>If you believe that the feature is available across multiple methods and should be added to the <code>deconvolute</code> interface, feel free to open an <a href="https://github.com/grst/immunedeconv/issues">issue</a> or pull request.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Finotello2018">
<p>Finotello, Francesca, and Zlatko Trajanoski. 2018. “Quantifying tumor-infiltrating immune cells from transcriptomics data.” <em>Cancer Immunology, Immunotherapy</em> 0. <a href="https://doi.org/10.1007/s00262-018-2150-z">https://doi.org/10.1007/s00262-018-2150-z</a>.</p>
</div>
<div id="ref-EPIC2017">
<p>Racle, Julien, Kaat de Jonge, Petra Baumgaertner, Daniel E Speiser, and David Gfeller. 2017. “Simultaneous enumeration of cancer and immune cell types from bulk tumor gene expression data.” <em>eLife</em> 6 (November): e26476. <a href="https://doi.org/10.7554/eLife.26476">https://doi.org/10.7554/eLife.26476</a>.</p>
</div>
<div id="ref-sturm2019">
<p>Sturm, Gregor, Francesca Finotello, Florent Petitprez, Jitao David Zhang, Jan Baumbach, Wolf H Fridman, Markus List, and Tatsiana Aneichyk. 2019. “Comprehensive Evaluation of Transcriptome-Based Cell-Type Quantification Methods for Immuno-Oncology.” <em>Bioinformatics</em> 35 (14): i436–i445.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gregor Sturm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
